File: python/pe004.py
def is_palindrome(n):
    return list(str(n)) == list(reversed(str(n)))
products = []
for i in range(100,1000):
    for j in range(100,1000):
        product = i * j
        if is_palindrome(product):
            products.append(product)
ans = max(products)
print(ans)

File: nim/pe004.nim
import math
import strutils
import unicode

proc pals(d: int): seq[int] =
  var ans: seq[int] = @[]

  for a in 10^(d-1)..<10^d:
    for b in a..<10^d:
      let c = a*b
      let d = intToStr(c)
      if d == reversed(d):
        ans.add(c)

  return ans


var ans = max(pals(3))
echo ans